src/ic-cdk/src/api/call.rs

```diff
@@ -68,11 +73,60 @@ impl Future for CallFuture {
 
     fn poll(self: Pin<&mut Self>, context: &mut Context<'_>) -> Poll<Self::Output> {
         let self_ref = Pin::into_inner(self);
+        let state_ptr = Arc::into_raw(self_ref.state.clone());
         let mut state = self_ref.state.write().unwrap();
 
         if let Some(result) = state.result.take() {
             Poll::Ready(result)
         } else {
+            // SAFETY:
+            // `callee`, being &[u8], is a readable sequence of bytes and therefore can be passed to ic0.call_new.
+            // `method`, being &str, is a readable sequence of bytes and therefore can be passed to ic0.call_new.
+            // `callback` is a function with signature (env : i32) -> () and therefore can be called as both reply and reject fn for ic0.call_new.
+            // `state_ptr` is a pointer created via Arc::into_raw, and can therefore be passed as the userdata for `callback`.
+            // `args`, being a &[u8], is a readable sequence of bytes and therefore can be passed to ic0.call_data_append.
+            // `cleanup` is a function with signature (env : i32) -> () and therefore can be called as a cleanup fn for ic0.call_on_cleanup.
+            // `state_ptr` is a pointer created via Arc::into_raw, and can therefore be passed as the userdata for `cleanup`.
+            // ic0.call_perform is always safe to call.
```

Comment:
CI is failing because the safety comments got mangled. The rule is that any *unsafe block* must be preceded by a comment describing why it's safe.

Reply:
I removed the comment, but didn't see any test failures related to the comment being present.

Reply:
Now CI started failing so I added the comment back just before the unsafe block. The comment used to say that "ic0.call_perform is always safe to call", but I wonder what that means given that `ic0.call_perform` can only be called  in the contexts `U Ry Rt T` (same for the other ic0 calls in the unsafe block).